source/common/modules/markdown-editor/linters/language-tool.ts

```diff
@@ -121,35 +144,24 @@ const ltLinter = linter(async view => {
     return [] // Hooray, nothing wrong!
   }
 
-  // Now, we have to remove those matches that are outside any textNode in the
-  // given document.
-  for (let i = 0; i < ltSuggestions.matches.length; i++) {
-    const from = ltSuggestions.matches[i].offset
-    const to = from + ltSuggestions.matches[i].length
-    let isValid = false
-
-    for (const node of textNodes) {
-      if (from >= node.from && to <= node.to) {
-        // As soon as we find a textNode that contains the match, we are good.
-        isValid = true
-        break
-      }
-    }
-
-    // Node is not valid --> remove
-    if (!isValid) {
-      ltSuggestions.matches.splice(i, 1)
-      i--
-    }
-  }
-
   // At this point, we have only valid suggestions that we can now insert into
   // the document.
+
+  // we want to filter out any words that are in our local dictionary
+  const localDictionary: string[] = await ipcRenderer.invoke('dictionary-provider', { command: 'get-user-dictionary' })
```

Comment:
This will have a heavy performance impact for large dictionaries. Recommendation: Keep it inside this file but outside the linter, and listen for update events from main. There are some other files that make use of the dictionary where I've implemented that.